include/deal.II/arborx/access_traits.h

```diff
@@ -0,0 +1,262 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_arborx_access_traits_h
+#define dealii_arborx_access_traits_h
+
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_WITH_ARBORX
+#  include <deal.II/base/bounding_box.h>
+
+#  include <ArborX.hpp>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ *  Structure that encapsulates a vector of dealii::Points used as queries.
+ */
+struct PointIntersect
```

Comment:
This and the next class are not really of general interest in deal.II. Could we move these into an internal namespace, maybe something like `internal::ArborX`?

Reply:
Users need to have access to these structures. This is how they build the `queries`

Reply:
So maybe in a namespace `dealii::ArborX`?

My concern is that the names of these class are quite generic. But they are not widely useful. The name (namespace) should somehow reflect the context in which they can be used.